Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionSetsWindow/MotionSetsWindowPlugin.cpp

```diff
@@ -393,12 +393,7 @@ namespace EMStudio
         }
 
         MotionSetsWindowPlugin* motionSetsPlugin = (MotionSetsWindowPlugin*)plugin;
-
-        // is the plugin visible? only update it if it is visible
-        if (motionSetsPlugin->GetDockWidget()->visibleRegion().isEmpty() == false)
```

Comment:
what's wrong with this logic

Reply:
There would be no problem with it, in case the selection model is clearly separated from the plugin/UI and could be separately updated. This is not the case though and the context menu in the motions window is relying on a fully up to date selection model in the motion set plugin. In case the motion set window is hidden while a motion set gets removed, the selection model would be invalid and that's why the incorrect context menu appears and leads to the reported crash.

The long-term solution is to separate the selection model but that is out of scope for the bug fix and we should reiterate on that thought after we removed the motion plugin (which will hopefully be happening in the shorter-/mid-term).